.github/workflows/reusable-san.yml

```diff
@@ -30,29 +33,42 @@ jobs:
       uses: actions/cache@v4
       with:
         path: config.cache
-        key: ${{ github.job }}-${{ env.IMAGE_OS_VERSION }}-${{ inputs.config_hash }}
+        key: ${{ github.job }}-${{ env.IMAGE_OS_VERSION }}-${{ inputs.sanitizer }}-${{ inputs.config_hash }}
     - name: Install dependencies
       run: |
         sudo ./.github/workflows/posix-deps-apt.sh
-        # Install clang-18
+        # Install clang
         wget https://apt.llvm.org/llvm.sh
         chmod +x llvm.sh
-        sudo ./llvm.sh 17  # gh-121946: llvm-18 package is temporarily broken
-        sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-17 100
-        sudo update-alternatives --set clang /usr/bin/clang-17
-        sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-17 100
-        sudo update-alternatives --set clang++ /usr/bin/clang++-17
-        # Reduce ASLR to avoid TSAN crashing
-        sudo sysctl -w vm.mmap_rnd_bits=28
-    - name: TSAN option setup
+
+        if [ "${SANITIZER}" = "TSan" ]; then
```

Comment:
Does this need to be conditional?

Should this go into an external script?

Reply:
Per discussion w/ Hugo, I'll not touch this place in the scope of this PR, but might look into it in a follow-up. Marking the thread as resolved.